core/src/main/kotlin/com/strumenta/kolasu/parsing/Parsing.kt

```diff
@@ -253,26 +266,37 @@ abstract class KolasuParser<R : Node, P : Parser, C : ParserRuleContext> {
     }
 
     fun parse(inputStream: InputStream, considerPosition: Boolean = true): ParsingResult<R> {
+        val time = System.currentTimeMillis()
         val code = inputStreamToString(inputStream)
         val result = parseFirstStage(code)
         var ast = parseTreeToAst(result.root!!, considerPosition)
         ast?.assignParents()
         val myIssues = result.issues.toMutableList()
         ast = if (ast == null) null else postProcessAst(ast, myIssues)
-        return ParsingResult(myIssues, ast, code, null)
+        return ParsingResult(myIssues, ast, code, null, result, System.currentTimeMillis() - time)
     }
 
     fun parse(code: String, considerPosition: Boolean = true): ParsingResult<R> {
+        val time = System.currentTimeMillis()
         val result = parseFirstStage(code)
         val ast = parseTreeToAst(result.root!!, considerPosition)
         ast?.assignParents()
-        return ParsingResult(result.issues, ast, code, null)
+        return ParsingResult(result.issues, ast, code, null, result, System.currentTimeMillis() - time)
     }
 
     fun parse(file: File, considerPosition: Boolean = true): ParsingResult<R> = parse(
         FileInputStream(file),
         considerPosition
     )
+
+    // For convenient use from Java
+    fun walk(node: Node) = node.walk()
+    @JvmOverloads
```

Comment:
I would add a blank line before the annotation

Reply:
OK